transport/socks5/stream_dialer.go

```diff
@@ -194,44 +199,68 @@ func (c *StreamDialer) DialStream(ctx context.Context, remoteAddr string) (trans
 	// buffer[2]: RSV
 	// buffer[3]: ATYP
 	if _, err = io.ReadFull(proxyConn, buffer[:4]); err != nil {
-		return nil, fmt.Errorf("failed to read connect server response: %w", err)
+		return nil, "", fmt.Errorf("failed to read connect server response: %w", err)
 	}
 
 	if buffer[0] != 5 {
-		return nil, fmt.Errorf("invalid protocol version %v. Expected 5", buffer[0])
+		return nil, "", fmt.Errorf("invalid protocol version %v. Expected 5", buffer[0])
 	}
 
 	// if REP is not 0, it means the server returned an error.
 	if buffer[1] != 0 {
-		return nil, ReplyCode(buffer[1])
+		return nil, "", ReplyCode(buffer[1])
 	}
 
-	// 4. Read address and length
+	// 4. Read BND.ADDR.
+	host := ""
 	var bndAddrLen int
 	switch buffer[3] {
```

Comment:
Let's introduce a readAddress as I mention in the packet code.

Reply:
I added `readAddress` and pushed the latest changes.